### PR TITLE
Mysql to Snowflake: Added include list.

### PR DESCRIPTION
**Pull request by HassanJaveed84**
Analytics Pipeline Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] If you have a migration please contact data engineering team before merging.
  - [ ] Before merging run full acceptance tests suite and provide URL for the acceptance tests run.
  - [ ] A member of data engineering team has approved the pull request.

### CODE REVIEW

File: edx/analytics/tasks/warehouse/load_internal_reporting_database.py [hunk 1]
```diff
@@ -840,9 +844,12 @@ def should_exclude_table(self, table_name):
...
         """
-        if any(re.match(pattern, table_name) for pattern in self.exclude):
-            return True
-        return False
+        if any(re.match(pattern, table_name) for pattern in self.include):
+            if any(re.match(pattern, table_name) for pattern in self.exclude):
+                return True
+            else:
+                return False
+        return True
```
Comment (brianhw) [inline edx/analytics/tasks/warehouse/load_internal_reporting_database.py:852]: Logic here doesn't quite look right to me.   If there is no INCLUDE being set, then I would want it to include everything, and only apply the exclude list.  That way we could merge this code and it wouldn't break the current settings. 

If it's easier, I'd also propose that we could ignore the EXCLUDE list if there is an INCLUDE list.   I'm not imagining that we really need to support both at the same time.  

So there would be an "if self.include:" clause, then an "elif self.exclude:" clause (or just a test of the "if any(re.match(pattern, table_name) for pattern in self.exclude):" directly).

**Review by brianhw — COMMENTED** [review]
I was also hoping that we could switch to using ImportMysqlDatabaseFromS3ToSnowflakeSchemaTask instead of ImportMysqlDatabaseToSnowflakeSchemaTask.   That would be a change in the jobs themselves.

File: edx/analytics/tasks/warehouse/load_internal_reporting_database.py (hunk 1 above)
Comment (HassanJaveed84) [reply edx/analytics/tasks/warehouse/load_internal_reporting_database.py:852]: Ah, I think I confused the requirements. I thought we wanted to limit it to whatever is in INCLUDE only.

Comment (HassanJaveed84) [reply edx/analytics/tasks/warehouse/load_internal_reporting_database.py:852]: This is from the A/C:
Add a new includes feature that:
* If anything is in the INCLUDE list, only these tables are copied (INCLUDE list takes precedence over exclude list)
* If EXCLUDE_FIELD is has value(s) it should still be applied to the items in the INCLUDES list

So the second point does mention of still applying the exclude.

Comment (HassanJaveed84) [reply edx/analytics/tasks/warehouse/load_internal_reporting_database.py:852]: But what you said makes sense. So this is what I'm going to do:

* Include everything if INCLUDE is empty. But still apply EXCLUDE.
* Include only INCLUDE if INCLUDE is set and ignore EXCLUDE altogether.

Is this what we want ?

Comment (brianhw) [reply edx/analytics/tasks/warehouse/load_internal_reporting_database.py:852]: I think so.  Note that the EXCLUDE_FIELD mentioned in the A/C is different than EXCLUDE, in that it deals with columns and not tables.

Comment (HassanJaveed84) [reply edx/analytics/tasks/warehouse/load_internal_reporting_database.py:852]: Thanks!
I think my brain casually chose to ignore _FIELD there in the A/C..

File: edx/analytics/tasks/warehouse/tests/test_internal_reporting_database.py [hunk 2]
```diff
@@ -153,3 +153,37 @@ def test_table_schema(self, mysql_query_results_mock):
...
             self.assertDictEqual(actual_schema, expected_schema)
+
+@ddt
+class ImportMysqlToSnowflakeTaskTest(TestCase):
+    """Test for ImportMysqlDatabaseToSnowflakeSchemaTask."""
+
+    def setUp(self):
+        self.task = ImportMysqlDatabaseToSnowflakeSchemaTask(
+            include=('auth_userprofile$', 'auth_user', 'assessment_assessmentpart', 'edxval_video'),
+            exclude=('auth_user$', 'assessment_assessment'),
```
Comment (pwnage101) [inline edx/analytics/tasks/warehouse/tests/test_internal_reporting_database.py:164]: Do you think it would add value to have two more tests: one with an empty exclude list, and another with an empty include list?
